### PR TITLE
Add intermediate checkpointing to `accelerate` trainers

### PR DESCRIPTION
**Pull request by jon-tow**
* Adds initial system for intermediate checkpointing into directories of the format: 

  `{TrainConfig.checkpoint_dir}/checkpoint_{step}`

Related Issue: #348

### CODE REVIEW

**Comment by LouisCastricato** [comment]
Oh hey a test trainer is a great addition!

File: trlx/trainer/accelerate_base_trainer.py [hunk 1]
```diff
@@ -487,7 +487,9 @@ def learn(self):  # noqa: C901
                     self.iter_count += 1
 
                     if self.iter_count % self.config.train.checkpoint_interval == 0:
-                        self.save()
+                        subfolder = f"checkpoint_{self.iter_count}"
```
Comment (cat-state) [inline trlx/trainer/accelerate_base_trainer.py:490]: can you zero left pad the iter count? will make the folders be visually sorted :)

File: trlx/trainer/accelerate_base_trainer.py [hunk 2]
```diff
@@ -533,7 +535,9 @@ def learn(self):  # noqa: C901
                     tbar.update()
 
                     if self.iter_count >= self.total_steps:
-                        self.save()
+                        subfolder = f"checkpoint_{self.iter_count}"
```
Comment (cat-state) [inline trlx/trainer/accelerate_base_trainer.py:538]: same as above

**Review by cat-state — APPROVED** [review]
LGTM aside from 1 tiny request

**Pull request merged**